awx/main/dispatch/worker/callback.py

```diff
@@ -26,7 +30,37 @@ class CallbackBrokerWorker(BaseWorker):
 
     MAX_RETRIES = 2
 
+    def __init__(self):
+        self.buff = {}
+        self.last_write = tz_now()
+
+    def read(self, queue):
+        try:
+            return queue.get(block=True, timeout=.25)
+        except QueueEmpty:
+            return 'FLUSH'
+
+    def flush(self):
+        should_flush = (
+            any([len(events) == 2000 for events in self.buff.values()]) or
+            (tz_now() - self.last_write).seconds > .1
```

Comment:
It might help to have these parameters of 0.25 seconds and 0.1 seconds as configurable settings, in case things go sideways in a complicated way we have not foreseen we can try toggling them.

Reply:
I think I might actually just make these match since they probably actually end up matching in practice anyways.  `.1` is probably performant enough to take advantage of bulk writes.  I'll test some more.